components/speedreader/speedreader_service.cc

```diff
@@ -6,33 +6,94 @@
 #include "brave/components/speedreader/speedreader_service.h"
 
 #include "base/feature_list.h"
+#include "base/metrics/histogram_macros.h"
 #include "brave/components/speedreader/features.h"
 #include "brave/components/speedreader/speedreader_pref_names.h"
+#include "brave/components/weekly_storage/weekly_storage.h"
 #include "components/prefs/pref_registry_simple.h"
 #include "components/prefs/pref_service.h"
 
 namespace speedreader {
 
+namespace {
+
+// Note: append-only enumeration! Never remove any existing values, as this enum
+// is used to bucket a UMA histogram, and removing values breaks that.
+constexpr std::array<uint64_t, 5> kSpeedReaderToggleBuckets{
+    0,   // 0
+    5,   // >0-5
+    10,  // 6-10
+    20,  // 11-20
+    30,  // 21-30
+         // 30+ => bucket 5
+};
+
+constexpr char kSpeedreaderToggleUMAHistogramName[] =
+    "Brave.SpeedReader.ToggleCount";
+
+constexpr char kSpeedreaderEnabledUMAHistogramName[] =
+    "Brave.SpeedReader.Enabled";
+
+void storeTogglesHistogram(uint64_t toggles) {
+  int bucket = 0;
+  for (const auto& bucketUpperBound : kSpeedReaderToggleBuckets) {
+    if (toggles > bucketUpperBound)
+      bucket = bucket + 1;
+  }
+
+  UMA_HISTOGRAM_EXACT_LINEAR(kSpeedreaderToggleUMAHistogramName, bucket, 5);
+}
+
+void storeSpeedReaderEnabledHistogram(PrefService* prefs_,
+                                      bool toggled,
+                                      bool enabledNow) {
+  if (toggled) {
```

Comment:
it seems to me that this function should be just split in two - one handling positive `toggled` case and the other

Reply:
and the other?

Reply:
Sorry I mean, one doing
```
    WeeklyStorage weekly(prefs_, kSpeedreaderPrefToggleCount);
    weekly.AddDelta(1);
    storeTogglesHistogram(weekly.GetWeeklySum());
```
and the other doing everything else

Reply:
per below discussions, want to:
1) report 0 count for toggles too, so not just storing for positive toggled case
2) "recently used" status relies on toggle count for this week

separation would be arbitrary and take away from clarity instead of adding to it